### ISSUE TITLE
Openshift route health checks are stuck in "Progressing" state

### ISSUE THREAD

**Issue by rhblind** [argoproj/argo-cd#8306]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version` (only json output, sorry).

**Describe the bug**

After upgrading Argo CD to `v2.2.2+unknown` routes are stuck in "Progressing" state.

**To Reproduce**

I have tried the following

- Deleted the route from Openshift
- Synced with Force, Prune and Replace
- Deleted the route from Argo and synced again

**Expected behavior**

The route health check state is "Healthy"

**Screenshots**

![MicrosoftTeams-image](https://user-images.githubusercontent.com/1574295/151531829-a669050d-2073-4953-b653-ea24ffe6293d.png)

**Version**

Unfortunately I don't have access to the server cli, but here's the json output from the Web interface.

```json
{
  "Version": "v2.2.2+unknown",
  "BuildDate": "2022-01-26T02:18:29Z",
  "GitTreeState": "clean",
  "GoVersion": "go1.16.12",
  "Compiler": "gc",
  "Platform": "linux/amd64",
  "KsonnetVersion": "unable to determine ksonnet version: exec: \"ks\": executable file not found in $PATH",
  "KustomizeVersion": "v4.2.0 1970-01-01T00:00:00Z",
  "HelmVersion": "v3.7.1+gbf78984",
  "KubectlVersion": "v0.22.2",
  "JsonnetVersion": "v0.17.0"
}
```

**Logs**

```
Paste any relevant application logs here.
```

**Comment by rhblind** [comment]
Tagging some co-workers for FYI reference
- @stianfro
- @niklashjh

**Comment by jannfis** [comment]
@ishitasequeira Can you please take a look at this?

**Comment by dnskr** [comment]
@rhblind Check release [v2.2.3](https://github.com/argoproj/argo-cd/releases/tag/v2.2.3)
```
Bug Fixes
-  fix: route health check stuck in 'Progressing' (#8170)
```

**Comment by rhblind** [comment]
@dnskr Ah, thank you very much! Missed that one!

**Comment by jannfis** [comment]
Fixed with #8170
